### PR TITLE
github/workflows: ditch softprops/turnstyle action

### PR DESCRIPTION
**Pull request by marquiz**
Use github's built-in concurrency, instead.

### CODE REVIEW

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -8,20 +8,15 @@ on:
...
-    - name: Turnstyle
-      uses: softprops/turnstyle@v1
-      with:
-        abort-after-seconds: 600
-        same-branch-only: false
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
     - name: Fetch gh-pages
       run: git fetch --no-tags --prune --depth=1 origin refs/heads/gh-pages:refs/heads/gh-pages
```
Comment (fmuyassarov) [inline .github/workflows/publish-docs.yml:22]: since we are here, should we perhaps replace this by https://github.com/actions/checkout ?

Comment (marquiz) [reply .github/workflows/publish-docs.yml:22]: That's an independent/unrelated change. Let's have a separate PR

Comment (fmuyassarov) [reply .github/workflows/publish-docs.yml:22]: ok

**Review by fmuyassarov — COMMENTED** [review]
LGTM
/cc @kad @askervin

**Review by fmuyassarov — APPROVED** [review]

**Review by kad — COMMENTED** [review]
/lgtm

**Review by kad — APPROVED** [review]

**Pull request merged**